handlers/vault.go

```diff
@@ -0,0 +1,122 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/gaia-pipeline/gaia/security"
+
+	"github.com/gaia-pipeline/gaia"
+	"github.com/labstack/echo"
+)
+
+type secret struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type updateSecret struct {
+	Key   string `json:"key"`
+	Value string `json:"newvalue"`
+}
+
+// UpdateSecret updates a secret using the vault.
+func UpdateSecret(c echo.Context) error {
+	s := new(updateSecret)
+	err := c.Bind(s)
+	if err != nil {
+		gaia.Cfg.Logger.Error("error reading secret", "error", err.Error())
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	v, err := security.NewVault()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error initializing vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	err = v.LoadSecrets()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error opening vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	v.Add(s.Key, []byte(s.Value))
+	err = v.SaveSecrets()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error saving vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	gaia.Cfg.Logger.Info("secret successfully updated")
+	return c.String(http.StatusOK, "secret successfully updated")
+}
+
+// AddSecret creates a secret using the vault.
+func AddSecret(c echo.Context) error {
+	s := new(secret)
+	err := c.Bind(s)
+	if err != nil {
+		gaia.Cfg.Logger.Error("error reading secret", "error", err.Error())
+		return c.String(http.StatusBadRequest, err.Error())
+	}
+	v, err := security.NewVault()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error initializing vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	err = v.LoadSecrets()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error opening vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	v.Add(s.Key, []byte(s.Value))
+	err = v.SaveSecrets()
+	if err != nil {
+		gaia.Cfg.Logger.Error("error saving vault", "error", err.Error())
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	gaia.Cfg.Logger.Info("secret successfully added")
+	return c.String(http.StatusOK, "secret successfully added")
```

Comment:
To be honest, I'm not really good at HTTP Status codes but this should be Created, right? 😆

Reply:
Ops, absolutely.